pkg/util/clock.go

```diff
@@ -137,6 +142,10 @@ func (f *FakeClock) HasWaiters() bool {
 	return len(f.waiters) > 0
 }
 
+func (f *FakeClock) Sleep(d time.Duration) {
+	f.Step(d)
```

Comment:
Happy to comment it. But: maybe the tests shouldn't be advancing the clock manually, and instead, sleeping the way normal non test scenarios do...

I actually wrote this so tests don't need to couple their impl to the clock mocking scheme ..... But maybe I'm not thinking of all the scenarios